array-manipulation-3/src/zip.js

```diff
@@ -0,0 +1,26 @@
+/* exported zip */
+
+// create a new array to hold changed data from input array
+// create a counter var to track the first array
+// loop the second array
+// if one array reaches it's length, return the first created array
+// as each pair of index's is accessed create a new array and store them in first-second order
+// add each new array to the first created array
+
+function zip(first, second) {
+  var newArr = [];
+  var f = 0;
+  for (let i = 0; i < second.length; i++) {
+    if (f === first.length || i === second.length) {
+      return newArr;
+    }
+    var newArr2 = [];
+    if (i === f) {
+      newArr2.push(first[f]);
+      newArr2.push(second[i]);
+      newArr.push(newArr2);
+      f++;
+    }
```

Comment:
Try simplifying this a little. Find the shorter length between the two arrays with Math.min(), then loop through the first array and push the pair of first at i and second at i into the new array.